source/write/bulk-write.txt

```diff
@@ -0,0 +1,287 @@
+.. _pymongo-bulk-write:
+
+=====================
+Bulk Write Operations
+=====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: insert, update, replace, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use bulk write operations with
+{+driver-short+}. You can perform multiple write operations
+in a single database call by using bulk operations.
+
+Consider a scenario in which you want to insert a document into a collection,
+update multiple other documents, then delete a document. Performing
+these operations by using their individual methods makes a call to the database
+for each method. You can instead use bulk write operations to perform
+these operations in a single database call.
+
+Performing Bulk Operations
+--------------------------
+
+To perform a bulk write operation, you must first create an instance of one of
+the following operation classes for each write operation you want to perform:
+
+- ``InsertOne``
+- ``UpdateOne``
+- ``UpdateMany``
+- ``ReplaceOne``
+- ``DeleteOne``
+- ``DeleteMany``
+
+Then, pass a list of these instances to the ``bulk_write()`` method.
+
+The following sections show how to create and use operation classes for
+different write operations.
+
+Insert Operations
+~~~~~~~~~~~~~~~~~
+
+To perform an insert operation, create an instance of ``InsertOne`` and specify
+the document you want to insert. To insert multiple documents, you must create
+an instance of ``InsertOne`` for each document.
+
+The following example creates an instance of ``InsertOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-insert-one
+   :end-before: end-bulk-insert-one
+   :language: python
+   :copyable:
+
+Update Operations
+~~~~~~~~~~~~~~~~~
+
+To perform a bulk update operation, create an instance of ``UpdateOne`` or
+``UpdateMany`` and pass it a **query filter** and the update operation you want
+to perform. A query filter specifies the criteria used to match documents to
+in your collection.
+
+``UpdateOne`` updates *the first* document that matches your query filter.
+``UpdateMany`` updates *all* documents that match your query filter.
+
+The following example creates an instance of ``UpdateOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-update-one
+   :end-before: end-bulk-update-one
+   :language: python
+   :copyable:
```

Comment:
S: Dividing this into two sections probably makes it easier to find what you're looking for at a glance. Maybe something like this (but feel free to edit)

```suggestion
To update a document, create an instance of ``UpdateOne`` and pass in
the following arguments:

- A query filter that specifies the document to update. ``UpdateOne`` updates *the first document* that matches this query filter.
- The update operation you want to perform [in what format].

The following example creates an instance of ``UpdateOne``:

.. literalinclude:: /includes/write/bulk-write.py
   :start-after: start-bulk-update-one
   :end-before: end-bulk-update-one
   :language: python
   :copyable:
   
   To update multiple documents, create an instance of ``UpdateMany``
   and pass in the same arguments. ``UpdateMany`` updates *all* documents
    that match your query filter.

```